custom_components/browser_mod/service.py

```diff
@@ -1,45 +1,51 @@
 import logging
 
-from homeassistant.helpers import device_registry
+from homeassistant.helpers import device_registry, template
+from homeassistant.util import dt as dt_util
+from homeassistant.exceptions import ServiceValidationError
 
 from .const import (
-    BROWSER_MOD_SERVICES,
+    BROWSER_MOD_BROWSER_SERVICES,
+    BROWSER_MOD_COMPONENT_SERVICES,
     DOMAIN,
     DATA_BROWSERS,
+    DATA_STORE,
 )
 
+from .browser import deleteBrowsers
+
 _LOGGER = logging.getLogger(__name__)
 
 
 async def async_setup_services(hass):
-    def call_service(service, targets, data):
-
-        browsers = hass.data[DOMAIN][DATA_BROWSERS]
-
-        # If no targets were specified, send to all browsers
-        if targets is None:
-            targets = browsers.keys()
-
-        for target in targets:
-            if target not in browsers:
-                continue
-            browser = browsers[target]
-            hass.create_task(browser.send(service, **data))
-
-    def handle_service(call):
-        service = call.service
-        data = {**call.data}
-
-        browsers = data.pop("browser_id", [])
+    def get_browser_ids(data, selectorSuffix=""):
+        browserIDSelector = "browser_id"
+        deviceIDSelector = "device_id"
+        areaIDSelector = "area_id"
+        if selectorSuffix:
+            browserIDSelector += f"_{selectorSuffix}"
+            deviceIDSelector += f"_{selectorSuffix}"
+            areaIDSelector += f"_{selectorSuffix}"
```

Comment:
This is unused (?), and could be hurtful if used.
Services can have a `target:` property - which actually used to be mandatory. That populates `device_id` and `area_id` with exactly those names.

Reply:
New `browser_mod.deregister_browser` service allows for `_exclude` IDs. So this is the way I wrote to be able to reuse code.

```
    # Hanldler for server browser_mod.deregister_browser
    async def deregister_browser(call):
        data = {**call.data}

        browserIDs = get_browser_ids(data)
        if browserIDs is None:
            browserIDs = set()
        
        excludedBrowserIDs = get_browser_ids(data, "exclude")
        if excludedBrowserIDs is None:
            excludedBrowserIDs = set()
....
```